### PR TITLE
Refactor jar in libs with gradle dependency (#296) 

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Following jars were successfully replaced with the gradle dependency:
```
commons-io-2.4.jar
gson-2.6.2.jar
kxml2-2.3.0.jar
opencsv-2.3.jar
slf4j-android-1.6.1-RC1.jar 
osmdroid-android-4.2
```

Removes following redundant library:
`guava-11.0.2`

### CODE REVIEW

**Comment by grzesiek2010** [comment]
You should start your branch from master. As you can see now there is an extra commit here.

File: collect_app/build.gradle [hunk 1]
```diff
@@ -96,6 +96,8 @@ task checkstyle(type: Checkstyle) {
     classpath = files()
 }
 
+
+
```
Comment (grzesiek2010) [inline collect_app/build.gradle:100]: Remove these unnecessary lines.

**Review by grzesiek2010 — CHANGES_REQUESTED** [review]

**Comment by grzesiek2010** [comment]
As I noticed guava library is unnecessary so I think you can remove this jar in this pr (but in separate commit) or am I wrong?

**Comment by grzesiek2010** [comment]
What about the others jars? Here is a dependency for for osmdroid for example: https://mvnrepository.com/artifact/org.osmdroid/osmdroid-android/4.2

**Comment by shobhitagarwal1612** [comment]
@grzesiek2010 Sorry for the extra commit. I will fix it.

**Comment by shobhitagarwal1612** [comment]
Also, the http-client is deprecated now, so its gradle dependency gives deprecated warning

> 
> Warning:WARNING: Dependency org.apache.httpcomponents:httpclient:4.0.3 is ignored for debug as it may be conflicting with the internal version provided by Android.
> Warning:WARNING: Dependency xpp3:xpp3:1.1.4c is ignored for debug as it may be conflicting with the internal version provided by Android.
> Warning:WARNING: Dependency org.apache.httpcomponents:httpclient:4.0.3 is ignored for debug as it may be conflicting with the internal version provided by Android.
> Warning:WARNING: Dependency xpp3:xpp3:1.1.4c is ignored for debug as it may be conflicting with the internal version provided by Android.
> Warning:WARNING: Dependency org.apache.httpcomponents:httpclient:4.0.3 is ignored for release as it may be conflicting with the internal version provided by Android.
> Warning:WARNING: Dependency xpp3:xpp3:1.1.4c is ignored for release as it may be conflicting with the internal version provided by Android.
> Warning:WARNING: Dependency org.apache.httpcomponents:httpclient:4.0.3 is ignored for release as it may be conflicting with the internal version provided by Android.
> Warning:WARNING: Dependency xpp3:xpp3:1.1.4c is ignored for release as it may be conflicting with the internal version provided by Android.

**Comment by shobhitagarwal1612** [comment]
For this library `osmbonuspack_v4.6.jar` I found out its gradle `'com.github.MKergall:osmbonuspack:6.2'` (https://github.com/MKergall/osmbonuspack). But this library uses minSdkVersion 10 and project's minSdkVersion is 9.
Since v4.6 is not available now so only alternative is to change minSdkVersion to 10. Also some of the methods used in v4.6 are deprecated in v6.2

**Comment by lognaturel** [comment]
Thank you very much, @shobhitagarwal1612! 😊 You did the right thing with leaving `http-client` and `osmbonuspack` as they are. Can you please copy your notes on both of them and any other interesting findings from libraries you couldn't move to maven control on #296? That will help us continue the process.

I'm also going to file a quick issue for style of the gradle file. That might be a good one for you to take @shobhitagarwal1612 since you've just been in that file so I'll tag you on it in case you want to do it.

**Pull request merged**